torch/distributed/_tensor/dispatch.py

```diff
@@ -174,6 +272,19 @@ def operator_dispatch(
         local_tensor_args = cast(Tuple[object, ...], local_tensor_args)
         local_tensor_kwargs = cast(Dict[str, object], local_tensor_kwargs)
         local_results = op_call(*local_tensor_args, **local_tensor_kwargs)
+        if (
+            (mesh is not None)
+            and (mesh.mesh.numel() < dist.get_world_size())
```

Comment:
hmmm I think this condition should be the same as the submesh condition? `mesh is not None and mesh.get_coordinate() is None`.

Reply:
`mesh.get_coordinate() is None` means that `self.rank` is not a part of submesh but this if-block is for participating ranks. The if-condition means "op is operating on a submesh (not a full mesh) and the current rank is a part of submesh". For a full mesh, we don't need to consider default value, right?

Reply:
for participating rank, shouldn't we check with `mesh.get_coordinate() is not None`?

Reply:
In this else-branch, `mesh.get_coordinate() is not None` has been checked:
```
if mesh is not None and mesh.get_coordinate() is None:
    ....
else:
    ....
    if (mesh is not None) and (mesh is a submesh):
        all_gather_object()
```

Reply:
Oh we want to check if the op is running in a "submesh" circumstance here, and only communicate when it returns a scalar value, got it, this make sense then.